### PR TITLE
Deprecate OffsetView constructors taking index_list_type

### PR DESCRIPTION
**Pull request by masterleinad**
While looking at https://github.com/kokkos/kokkos/pull/4805, I found a `OffsetView` constructor taking `index_list_type`,i.e., `std::initializer_list`, objects as argument to denote begin and end for every dimension. This pull request deprecates that behavior to use `std::pair` instead so that specifying too many arguments is not possible.

### CODE REVIEW

File: containers/src/Kokkos_OffsetView.hpp [hunk 1]
```diff
@@ -41,6 +41,7 @@ template <typename iType,
 using IndexRange = Kokkos::Array<iType, 2>;
 
 using index_list_type = std::initializer_list<int64_t>;
+using range_type      = std::pair<int64_t, int64_t>;
```
Comment (dalg24) [inline containers/src/Kokkos_OffsetView.hpp:44]: I am not sure about adding that guy.  I am surprised we defined `index_list_type` like that in the first place.

**Review by dalg24 — COMMENTED** [review]
Looks fine beside the adding `Kokkos::Experimental::range_type`

**Comment by masterleinad** [comment]
> Looks fine beside the adding Kokkos::Experimental::range_type

Do you prefer to use `std::pair` directly instead of the alias then?

**Comment by dalg24** [comment]
As long we are not injecting an alias in the namespace.
Note that you could write your constructor to take anything that supports `std::get<{0,1}>(R const&)` with `std::tuple_size<R>::value==2`

**Comment by masterleinad** [comment]
> Note that you could write your constructor to take anything that supports `std::get<{0,1}>(R const&)` with `std::tuple_size<R>::value==2`

I prefer to have a constructor that allows me to do 
```
offset_view_type("o1", {-1, 3}, {-2,2});
```
and I don't see a need to introduce another templated one taking other tuple-like containers yet.

**Review by dalg24 — CHANGES_REQUESTED** [review]
Comment about the design choices above the constructor.

**Review by dalg24 — APPROVED** [review]

**Comment by masterleinad** [comment]
Retest this please.

**Review by Rombur — APPROVED** [review]

**Review by crtrott — APPROVED** [review]

**Pull request merged**